src/pipecat/services/aws_nova_sonic/aws.py

```diff
@@ -0,0 +1,1002 @@
+        #     print(f"[pk] assistant content start: {content}")
+
+        if content.role == Role.ASSISTANT:
+            if content.type == ContentType.AUDIO:
+                # Note that an assistant response can comprise of multiple audio blocks
+                if not self._assistant_is_responding:
+                    # The assistant has started responding.
+                    self._assistant_is_responding = True
+                    await self._report_assistant_response_started()
+
+    async def _handle_text_output_event(self, event_json):
+        # This should never happen
+        if not self._content_being_received:
+            return
+        content = self._content_being_received
+
+        text_content = event_json["textOutput"]["content"]
+        # print(f"[pk] text output. content: {text_content}")
+        # if content.role == Role.ASSISTANT:
+        #     print(f"[pk] assistant text output. content: {text_content}")
+
+        # Bookkeeping: augment the current content being received with text
+        # Assumption: only one text content per content block
+        content.text_content = text_content
+
+    async def _handle_audio_output_event(self, event_json):
+        # This should never happen
+        if not self._content_being_received:
+            return
+
+        # Get audio
+        audio_content = event_json["audioOutput"]["content"]
+        # print(f"[pk] audio output. content: {len(audio_content)}")
+
+        # Push audio frame
+        audio = base64.b64decode(audio_content)
+        # TODO: make sample rate + channels (used in multiple places) consts
+        frame = TTSAudioRawFrame(
+            audio=audio,
+            sample_rate=24000,
+            num_channels=1,
+        )
+        await self.push_frame(frame)
+
+    async def _handle_tool_use_event(self, event_json):
+        # This should never happen
+        if not self._content_being_received:
+            return
+
+        # Get tool use details
+        tool_use = event_json["toolUse"]
+        function_name = tool_use["toolName"]
+        tool_call_id = tool_use["toolUseId"]
+        arguments = json.loads(tool_use["content"])
+
+        # print(
+        #     f"[pk] tool use - function_name: {function_name}, tool_call_id: {tool_call_id}, arguments: {arguments}"
+        # )
+
+        # Call tool function
+        if self.has_function(function_name):
+            if function_name in self._functions.keys():
+                await self.call_function(
+                    context=self._context,
+                    tool_call_id=tool_call_id,
+                    function_name=function_name,
+                    arguments=arguments,
+                )
+            elif None in self._functions.keys():
+                await self.call_function(
+                    context=self._context,
+                    tool_call_id=tool_call_id,
+                    function_name=function_name,
+                    arguments=arguments,
+                )
```

Comment:
Can we remove this if ? It looks like we are invoking the same code in both:
```
await self.call_function(
    context=self._context,
    tool_call_id=tool_call_id,
    function_name=function_name,
    arguments=arguments,
)
```

Reply:
Fixed (ini both places)

Reply:
Oh, I see, Yeah, I think you are right, the single if should be enough.